server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/model/MockSMTPBehaviorInformationTest.java

```diff
@@ -0,0 +1,179 @@
+        EqualsVerifier.forClass(MockSMTPBehaviorInformation.class)
+            .verify();
+    }
+
+    @Nested
+    class UnlimitedAnswersTest {
+
+        @Test
+        void equalsShouldAlwaysBeTrue() {
+            EqualsVerifier.forClass(MockSMTPBehaviorInformation.UnlimitedAnswersCounter.class)
+                .verify();
+        }
+
+        @Test
+        void isRemainingShouldReturnTrue() {
+            RemainingAnswersCounter unlimited = new UnlimitedAnswersCounter();
+
+            assertThat(unlimited.hasRemainingAnswers())
+                .isTrue();
+        }
+
+        @Test
+        void isRemainingShouldReturnTrueAfterDecreasing() {
+            RemainingAnswersCounter unlimited = new UnlimitedAnswersCounter();
+            unlimited.decrease();
+            unlimited.decrease();
+            unlimited.decrease();
+
+            assertThat(unlimited.hasRemainingAnswers())
+                .isTrue();
+        }
+
+        @Test
+        void remainedShouldReturnEmptyOptional() {
+            RemainingAnswersCounter unlimited = new UnlimitedAnswersCounter();
+
+            assertThat(unlimited.getValue())
+                .isEmpty();
+        }
+
+        @Test
+        void remainedShouldReturnEmptyOptionalAfterDecreasing() {
+            RemainingAnswersCounter unlimited = new UnlimitedAnswersCounter();
+            unlimited.decrease();
+            unlimited.decrease();
+
+            assertThat(unlimited.getValue())
+                .isEmpty();
+        }
+    }
+
+    @Nested
+    class LimitedAnswersTest {
+
+        @Test
+        void shouldMatchBeanContract() {
+            EqualsVerifier.forClass(MockSMTPBehaviorInformation.LimitedAnswersCounter.class)
+                .suppress(Warning.NULL_FIELDS)
+                .verify();
+        }
+
+        @Test
+        void constructorShouldThrowWhenPassingNegativeAnswersCount() {
+            assertThatThrownBy(() -> new LimitedAnswersCounter(-100))
+                .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        void constructorShouldThrowWhenPassingZeroAnswersCount() {
+            assertThatThrownBy(() -> new LimitedAnswersCounter(0))
+                .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        void isRemainingShouldReturnTrueWhenNoDecrease() {
+            RemainingAnswersCounter limited = new LimitedAnswersCounter(1);
+
+            assertThat(limited.hasRemainingAnswers())
+                .isTrue();
+        }
+
+        @Test
+        void isRemainingShouldReturnTrueWhenNotEnoughDecremental() {
+            RemainingAnswersCounter limited = new LimitedAnswersCounter(3);
+            limited.decrease();
+            limited.decrease();
+
+            assertThat(limited.hasRemainingAnswers())
+                .isTrue();
+        }
+
+        @Test
+        void isRemainingShouldReturnFalseWhenEnoughDecremental() {
+            RemainingAnswersCounter limited = new LimitedAnswersCounter(3);
```

Comment:
I prefer a `for` and a variable

Reply:
I don't like for, not so many lines can save but harder code to read

Reply:
having a variable would make the causality link explicit